src/discussions/posts/PostsList.jsx

```diff
@@ -51,10 +52,11 @@ function PostsList({
       isFilterChanged,
     };
 
+    if ((prevFilters.current.search !== '' && prevFilters.current.search !== filters.search) || !showOwnPosts) {
+      dispatch(fetchThreads(courseId, params));
+    }
```

Comment:
This part of the condition `(prevFilters.current.search !== '' && prevFilters.current.search !== filters.search)` would be false always. `prevFilters` is initialized with an empty string and does not change after that. So its value remains an empty string and this check `prevFilters.current.search !== ''` will get false. 
Indirectly,  Your check only depends on `!showOwnPosts`. The previous code was working on the same logic. However, current change is not required.